### PR TITLE
Update HTTP based clients to send 'x-goog-api-client' headers.

### PR DESCRIPTION
**Pull request by landrito**
Don't merge unless @bjwatson LGTMS.

### CODE REVIEW

**Comment by blowmage** [comment]
Should DNS and Resource Manager also get these changes?

**Comment by landrito** [comment]
Those weren't on my radar. One sec, let me add them.

**Comment by bjwatson** [comment]
Is CircleCI failing due to flaky tests?

**Comment by blowmage** [comment]
The build is failing due to a code style violation. One of the lines changes has 81 characters, which is over the 80 character limit for lines.

File: google-cloud-resource_manager/lib/google/cloud/resource_manager/service.rb [hunk 1]
```diff
@@ -43,6 +43,9 @@ def initialize credentials, retries: nil, timeout: nil
           @service.request_options.retries = retries || 3
           @service.request_options.timeout_sec      = timeout
           @service.request_options.open_timeout_sec = timeout
+          @service.request_options.header ||= {}
+          @service.request_options.header["x-goog-api-client"] = \
+            "gl-ruby/#{RUBY_VERSION} gccl/#{Google::Cloud::ResourceManager::VERSION}"
```
Comment (blowmage) [inline google-cloud-resource_manager/lib/google/cloud/resource_manager/service.rb]: @landrito This line is too long and is the reason for the failed CI builds.

Comment (landrito) [inline google-cloud-resource_manager/lib/google/cloud/resource_manager/service.rb]: Done!

Comment (bjwatson) [inline google-cloud-resource_manager/lib/google/cloud/resource_manager/service.rb]: Awesome, thanks!

**Pull request merged**